[RFC] Disable bold text bright highlighting in terminal

choco opened the pull request:
It makes more sense to let the user application terminal emulator
decide how to render bold text.

Fixes https://github.com/neovim/neovim/issues/2453

justinmk commented on src/nvim/terminal.c:
```diff
@@ -201,7 +201,7 @@ Terminal *terminal_open(TerminalOptions opts)
   // Setup state
   VTermState *state = vterm_obtain_state(rv->vt);
-  vterm_state_set_bold_highbright(state, true);
+  vterm_state_set_bold_highbright(state, false);
```
Shouldn't this line be deleted then?

cHoco replied:
Yeah we could, since as a default libvterm initializes the state of this variable to false. I just thought about keeping it in case libvterm changes its defaults (unlikely). Let me know what you prefer :)

Pyrohh replied:
I don't think keeping it is needed.

If the default is changed, it will most likely be mentioned in the release notes, which are generally very thorough given how few commits I've noticed between releases.

choco committed:
Disable bold text bright highlighting in terminal

It makes more sense to let the user application terminal emulator
decide how to render bold text.

Pyrohh commented:
This fixes the issue described #2453 for me (https://github.com/neovim/neovim/issues/2453#issuecomment-97934231).

cHoco commented:
Ok, removed the line :+1: 
Taking a look again at issue https://github.com/neovim/neovim/issues/2453 I notices there was also this PR https://github.com/neovim/neovim/issues/2996 solving this problem, but I don't think it makes sense to make this option configurable from the user pov.

Pyrohh commented:
> I don't think it makes sense to make this option configurable from the user pov

I agree; the extra complexity isn't worth it IMO, and as @justinmk said, I'm not excited about perpetuating `g:*` variables to control core features.

Besides that, the CLA still hasn't been signed by that user, so I'll close said PR.

justinmk commented:
:+1:

justinmk commented:
Merged

Pyrohh commented:
Thanks @cHoco and @justinmk.

cHoco commented:
Thanks to you guys, for all the amazing work that you do for this project :+1: you're the real MVPs :)
